sqlobject/src/main/java/org/jdbi/v3/sqlobject/customizers/Define.java

```diff
@@ -37,15 +38,27 @@
      * The key for the attribute to set. The value will be the value passed to the annotated argument
      * @return the attribute key
      */
-    String value();
+    String value() default "";
 
     class Factory implements SqlStatementCustomizerFactory
     {
         @Override
-        public SqlStatementCustomizer createForParameter(Annotation annotation, Class<?> sqlObjectType, Method method, final Object arg)
+        public SqlStatementCustomizer createForParameter(Annotation annotation, Class<?> sqlObjectType, Method method, Parameter param, final Object arg)
         {
-            Define d = (Define) annotation;
-            final String key = d.value();
+            Define define = (Define) annotation;
+
+            String name = define.value();
+            if (name.isEmpty()) {
+                if (param.isNamePresent()) {
+                    name = param.getName();
+                } else {
+                    throw new UnsupportedOperationException("A @Define parameter was not given a name, "
```

Comment:
Not without a lot of trouble--this is configured in the POM. We'd have to spin up a separate artifact with a different compiler configuration just to test this. I'm not sure it's worth it.